app/controllers/application_controller.rb

```diff
@@ -3,9 +3,9 @@ class ApplicationController < ActionController::Base
 
   before_action :configure_permitted_parameters, if: :devise_controller?
 
-  helper_method :is_same_user
+  helper_method :authorize_user
```

Comment:
this will never work as `helper_method` because of the redirection. I would create a helper method (in a helper file) just for view stuff.

As an extra, you could try to use an authorization gem https://github.com/elabs/pundit (ONLY IF YOU HAVE TIME!)